tests/api_test.py

```diff
@@ -1736,14 +1736,21 @@ def f(x):
           d = add a 2.0
           e = cond[ false_jaxpr={ lambda  ; b a.
                                   let c = sub a b
-                                  in [c] }
+                                  in (c,) }
                     linear=(False, False, False, False)
                     true_jaxpr={ lambda  ; b a.
                                  let c = add a b
-                                 in [c] } ] b a c a d
-      in [e] }
+                                 in (c,) } ] b a c a d
+      in (e,) }
         """)
 
+  def test_pmap_constants_not_replicated(self):
+    x = jax.random.uniform(jax.random.PRNGKey(0), shape=(1000, 10))
+    f = jax.pmap(lambda i: x[i])
+    s = jax.xla_computation(f)(np.arange(10)).GetHloText()
+    self.assertIn('1000,10', s)
+    self.assertNotIn('1,1000,10', s)
```

Comment:
Wouldn't this be replicated 10,1000,10, with the first 10 being the leading axis? 

This test is hard to understand. I think it would be easier if we assert on more context around '1000,10'?

Reply:
This is a typo on my part, and I'm confused why it isn't just crashing! I meant to give it `np.arange(1)`, so this test would run without multiple devices available.

Reply:
I'll revise the checks to have more context.

Reply:
I added more context to the test. As to why the pmap isn't failing when mapping over too large an axis, I believe it's because we've set up `jax.xla_computation` to work even when building computations for more devices than are available.